container/writer.go

```diff
@@ -70,6 +72,14 @@ func NewWriter(writer io.Writer, codec Codec, recordsPerBlock int64) (*Writer, e
 		avroWriter.compressedWriter = avroWriter.blockBuffer
 	}
 
+	if len(schema) == 1 {
+		avroWriter.headerWritten = true
```

Comment:
I think it would make sense to move setting `avorWriter.headerWritten` into `writerHeader` itself, so it isn't duplicated.